server/src/main/java/io/deephaven/server/console/ScopeTicketResolver.java

```diff
@@ -58,29 +51,29 @@ public SessionState.ExportObject<Flight.FlightInfo> flightInfoFor(
         // there is no mechanism to wait for a scope variable to resolve; require that the scope variable exists now
         final String scopeName = nameForDescriptor(descriptor, logId);
 
-        final ScriptSession gss = scriptSessionProvider.get();
-        final Flight.FlightInfo flightInfo =
-                gss.getExecutionContext().getUpdateGraph().sharedLock().computeLocked(() -> {
-                    Object scopeVar = gss.getVariable(scopeName, null);
-                    if (scopeVar == null) {
-                        throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
-                                "Could not resolve '" + logId + ": no variable exists with name '" + scopeName + "'");
-                    }
-                    if (scopeVar instanceof Table) {
-                        scopeVar = authorization.transform(scopeVar);
-                        return TicketRouter.getFlightInfo((Table) scopeVar, descriptor, flightTicketForName(scopeName));
-                    }
+        QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        Object scopeVar = queryScope.readParamValue(scopeName, null);
+        if (scopeVar == null) {
+            throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
+                    "Could not resolve '" + logId + ": no variable exists with name '" + scopeName + "'");
+        }
+        if (!(scopeVar instanceof Table)) {
+            throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
+                    "Could not resolve '" + logId + "': no variable exists with name '" + scopeName + "'");
+        }
 
-                    throw Exceptions.statusRuntimeException(Code.NOT_FOUND,
-                            "Could not resolve '" + logId + "': no variable exists with name '" + scopeName + "'");
-                });
+        Table transformed = authorization.transform((Table) scopeVar);
+        Flight.FlightInfo flightInfo =
+                TicketRouter.getFlightInfo(transformed, descriptor, flightTicketForName(scopeName));
 
         return SessionState.wrapAsExport(flightInfo);
     }
 
     @Override
     public void forAllFlightInfo(@Nullable final SessionState session, final Consumer<Flight.FlightInfo> visitor) {
-        scriptSessionProvider.get().getVariables().forEach((varName, varObj) -> {
+        QueryScope queryScope = ExecutionContext.getContext().getQueryScope();
+        queryScope.getParamNames().forEach((varName) -> {
+            Object varObj = queryScope.readParamValue(varName);
```

Comment:
Same atomicity issue as in `Sql`.

Reply:
True, but not a new issue. Do we want a method that returns a map<string,value>?

Reply:
It looks like you implemented that method, but didn't apply it to these comments.